monitoring/harmony-monitor/blockchain-watchdog/reporting-server.go

```diff
@@ -224,9 +227,46 @@ func (m *monitor) produceCSV(w http.ResponseWriter, req *http.Request) {
 		case blockHeaderReport:
 			filename = blockHeaderReport + ".csv"
 			records = append(records, headerInformationCSVHeader)
+
+			shard, ex := req.URL.Query()["shard"]
+			if !ex {
+				http.Error(w, "shard not chosen in query param", http.StatusBadRequest)
+				return
+			}
+			for _, v := range m.SummarySnapshot[headerSumry][shard[0]].(any)["records"].([]interface{}) {
```

Comment:
slightly nervous about this line, lots of chances for it to explode (the `.(any)`, the `.([]interface{})`. And why hardcoded to `shard[0]`

Reply:
1. Maybe we should rework the Summary struct to be less generic to avoid the `.(any)`, etc casts in order to access the struct.
2. URL.Query returns a slice, so need to access index 0 of shard.

Reply:
Fair - 2 is fine then since we are the callers anyway (from html report), for 1 let's think about that a bit more later.